lib/HandlerBase.h

```diff
@@ -44,11 +44,9 @@ class HandlerBase {
 
     void start();
 
-    /*
-     * get method for derived class to access weak ptr to connection so that they
-     * have to check if they can get a shared_ptr out of it or not
-     */
-    ClientConnectionWeakPtr getCnx() const { return connection_; }
+    ClientConnectionWeakPtr getCnx() const;
```

Comment:
nit: for later, we could return the `shared_ptr` here, already locked inside the method

Reply:
Good suggestion. I will do it later